File: Linked_List/linkedList.js
class Node{
    constructor(data){
        this.data = data
        this.next = null
    }
}

class LinkedList{
    constructor(){
        this.head = null
    }

    insertAtLast(data){
        let node = new Node(data)


        // Our Linked List is Empty
        if(this.head == null){
            this.head = node
        }
        else{
            // We have the data in our linked list

            let currNode = this.head

            while(currNode.next != null){
                currNode = currNode.next
            }

            currNode.next = node
        }
    }

    deleteNode(dataKey){
        let prevNode = null
        let currNode = this.head

        // if we have to delete the data from head

        if(currNode.data == dataKey){
            currNode = currNode.next
            this.head = currNode
            return
        }
        
        while(currNode != null){
            if(currNode.data == dataKey){
                prevNode.next = currNode.next
                return
            }
            prevNode = currNode
            currNode = currNode.next
        }
        
        console.log(`Node With data not found`)
    }

    display(){
        let currNode = this.head

        let linkedList = ''

        while(currNode != null){
            if(currNode.next != null)
                linkedList += currNode.data + ' -> '
            else linkedList += currNode.data

            currNode = currNode.next
        }

        return linkedList
    }
}

linkedList = new LinkedList()

for(let i = 0;i<10;i++){
    linkedList.insertAtLast(i)
}

console.log(linkedList.display())

// from mid
linkedList.deleteNode(2)
console.log(linkedList.display())

// from head
linkedList.deleteNode(0)
console.log(linkedList.display())

// from last
linkedList.deleteNode(9)
console.log(linkedList.display())

File: Linked_List/Singly Linked List/LinkedList.c
#include <stdio.h>
#include <stdlib.h>
#include <conio.h>

typedef struct node
{
    int data;
    struct node *next;
} node;

typedef struct linkedList
{
    struct node *head;
} linkedList;

node *insertAtLast(node *head, int data)
{
    node *newnode;
    newnode = (node *)malloc(sizeof(node));
    newnode->data = data;
    newnode->next = NULL;
    if (head == NULL)
    {
        return newnode;
    }
    node *currPtr = head;

    while (currPtr->next != NULL)
    {
        currPtr = currPtr->next;
    }

    currPtr->next = newnode;

    return head;
}

void deleteNode(node **head, int dataKey)
{
    node *prevNode = NULL;
    node *currNode = *head;

    if (currNode->data == dataKey)
    {
        node *deleteNode = currNode;
        currNode = currNode->next;
        *head = currNode;
        free(deleteNode);
        return;
    }

    while (currNode != NULL)
    {
        if (currNode->data == dataKey)
        {
            node *deleteNode = currNode;
            prevNode->next = currNode->next;
            free(deleteNode);
            return;
        }

        prevNode = currNode;
        currNode = currNode->next;
    }

    printf("Node with Data Not Found");
}

void display(node *head)
{
    node *currPtr = head;

    while (currPtr != NULL)
    {
        if (currPtr->next != NULL)
            printf("%d -> ", currPtr->data);
        else
            printf("%d", currPtr->data);

        currPtr = currPtr->next;
    }
}

int main()
{
    linkedList ll;
    ll.head = NULL;

    for (int i = 0; i < 10; i++)
    {
        ll.head = insertAtLast(ll.head, i);
    }

    display(ll.head);
    printf("\n");

    // Deletion from head
    deleteNode(&ll.head, 0);
    display(ll.head);
    printf("\n");

    // Deletion from mid
    deleteNode(&ll.head, 5);
    display(ll.head);
    printf("\n");

    // Deletion from last
    deleteNode(&ll.head, 9);
    display(ll.head);
    printf("\n");
    return 0;
}